source/dub/project.d

```diff
@@ -354,7 +354,10 @@ class Project {
 					}
 				} else if (m_selections.hasSelectedVersion(basename)) {
 					vspec = m_selections.getSelectedVersion(basename);
-					if (vspec.path.empty) p = m_packageManager.getBestPackage(dep.name, vspec);
+					if (!vspec.repository.empty) {
```

Comment:
the precedence is probably wrong here. Everywhere else before you first checked path, then repository. Here you first check repository.